component/main.jsonnet

```diff
@@ -28,8 +28,205 @@ local net_pol = kube.NetworkPolicy('allow-from-waf-namespaces') {
   },
 };
 
+local rules_crd_gateway = {
+  apiGroups: [ 'apiextensions.k8s.io' ],
+  resources: [ 'customresourcedefinitions' ],
+  resourceNames: [
+    'backendtlspolicies.gateway.networking.k8s.io',
+    'gatewayclasses.gateway.networking.k8s.io',
+    'gateways.gateway.networking.k8s.io',
+    'grpcroutes.gateway.networking.k8s.io',
+    'httproutes.gateway.networking.k8s.io',
+    'referencegrants.gateway.networking.k8s.io',
+    'tcproutes.gateway.networking.k8s.io',
+    'tlsroutes.gateway.networking.k8s.io',
+    'udproutes.gateway.networking.k8s.io',
+  ],
+  verbs: [ 'get' ],
+};
+local rules_resources_gateway_ronly = {
+  apiGroups: [ 'gateway.networking.k8s.io' ],
+  resources: [
+    'gatewayclasses',
+  ],
+  verbs: [ 'get', 'list', 'watch' ],
+};
+local rules_resources_gateway_read = rules_resources_gateway_ronly {
+  apiGroups: [ 'gateway.networking.k8s.io' ],
+  resources: [
+    'backendtlspolicies',
+    'gateways',
+    'grpcroutes',
+    'httproutes',
+    'referencegrants',
+    'tcproutes',
+    'tlsroutes',
+    'udproutes',
+  ],
+  verbs: [ 'get', 'list', 'watch' ],
+};
```

Comment:
why split between `gatewayclasses` and the rest? Is it actually the rest? why not `resources: ['*']` for the second rule definition?

Reply:
ah right... never mind then.